scripts/integration_test.py

```diff
@@ -0,0 +1,148 @@
+import random
+from concurrent.futures import ThreadPoolExecutor
+import concurrent.futures
+from io import StringIO
+
+import numpy as np
+import pandas as pd
+import requests
+from urllib.error import HTTPError
+
+X_SOURCE = 'API de Series de Tiempo: Test de Integración'
+
+
+def read_source_csv(serie_id: str, metadata: pd.DataFrame):
+    serie_metadata = metadata.loc[serie_id, :]
+    if metadata is None:
+        return None
+
+    download_url = serie_metadata.distribucion_url_descarga
+    title = serie_metadata.serie_titulo
+    try:
+        csv = pd.read_csv(download_url, parse_dates=['indice_tiempo'], index_col='indice_tiempo')
+    except HTTPError:
+        return None
+
+    return csv[[title]]
+
+
+def get_equality_array(api_df: pd.DataFrame, original_df: pd.DataFrame):
+    df = pd.merge(api_df, original_df, left_index=True, right_index=True)
+    api, original = df.columns
+    equality = np.isclose(df[api], df[original], rtol=0.001, equal_nan=True)
+    return equality
+
+
+def get_series_metadata(api_url: str):
+    metadata_endpoint = f'{api_url}series/api/dump/series-tiempo-metadatos.csv'
+    metadata = pd.read_csv(metadata_endpoint).set_index('serie_id')
+    return metadata
+
+
+class IntegrationTest:
+
+    def __init__(self, series_metadata: pd.DataFrame, fetcher, threads=10):
+        """Test de integración de la API de series. A partir de los metadatos de las series
+        (obtenibles a partir de un dump de metadatos), se consultan los datos de todas las
+        series una vez, y se comparan con los datos originales.
+        """
+        self.series_metadata = series_metadata
+        self.fetcher = fetcher
+        self.errors = []
+        self.threads = threads
+
+    def test(self):
+        self.run_concurrent()
+        return self.errors
+
+    def run_concurrent(self):
+        with ThreadPoolExecutor(max_workers=self.threads) as executor:
+            futures = []
+            for serie_id in self.series_metadata.index:
+                fs = executor.submit(self.test_serie, serie_id)
+                futures.append(fs)
+            concurrent.futures.wait(futures)
+
+    def test_serie(self, serie_id):
+        api_df = self.read_api_csv(serie_id, last=1000)
```

Comment:
Ya que más abajo tenés un `if api_df is None -> Return`, se podría hacer acá mismo el chequeo de si es None y, en caso de que lo sea, te ahorrás ejecutar `read_source_csv()`.